app/assets/javascripts/action_tracker.js

```diff
@@ -0,0 +1,118 @@
+
+"use strict";
+
+var ActionTracker = function () {
+
+  var storage,
+      callbacks = {};
+
+  function constructor() {
+    storage = new Storage();
+  }
+
+  function setCallbacks(callbacksObj) {
+    callbacks = callbacksObj;
+  }
+
+  function start(list) {
+    storage.queue(list);
+  }
+
+  function process() {
+    while(typeof storage.getFirst() !== 'undefined') {
+      var tracker = new Tracker(storage.getFirst());
+      tracker.send();
```

Comment:
This send doesn't really ensure that the item popped from the storage, right? Why you still need to get it then dequeue it?